### DISCUSSION TITLE
How filament registering itself as blade component namespace e.g. <x-filament::dropdown>

### DISCUSSION THREAD

**Discussion by khanakia** [filamentphp/filament#8380] [Help] [answered]
Based on Laravel Blade Component docs one needs to register their package as a Component namespace 

https://laravel.com/docs/10.x/blade#manually-registering-package-components
```php
Blade::componentNamespace('Nightshade\\Views\\Components', 'nightshade');
```

But i was digging through filament code and could not find any code like that so how filament is registering the component director`packages/support/resources/views/components` under the `x-filament` namepace?

**Comment by pboivin** [comment, marked as answer]
I think this is done automatically through https://github.com/spatie/laravel-package-tools

For example: https://github.com/filamentphp/filament/blob/3.x/packages/forms/src/FormsServiceProvider.php#L17-L24

**Reply by khanakia** [reply]
Thanks, It makes sense now. As creator used this 3rd party package to create a filament package that's why I could not find the Blade Namespace code.
